samcli/cli/types.py

```diff
@@ -9,15 +9,47 @@
 import click
 
 
-def _value_regex(delim):
-    return f'(\\"(?:\\\\.|[^\\"\\\\]+)*\\"|(?:\\\\.|[^{delim}\\"\\\\]+)+)'
+def _match_regex(match, delim):
+    return f'(\\"(?:\\\\{match}|[^\\"\\\\]+)*\\"|(?:\\\\{match}|[^{delim}\\"\\\\]+)+)'
 
 
 KEY_REGEX = '([A-Za-z0-9\\"]+)'
+# Tags have additional constraints and they allow "+ - = . _ : / @" apart from alpha-numerics.
+TAG_REGEX = '[A-Za-z0-9\\"_:\\.\\/\\+-\\@=]'
+
 # Use this regex when you have space as delimiter Ex: "KeyName1=string KeyName2=string"
-VALUE_REGEX_SPACE_DELIM = _value_regex(" ")
+VALUE_REGEX_SPACE_DELIM = _match_regex(match=".", delim=" ")
 # Use this regex when you have comma as delimiter Ex: "KeyName1=string,KeyName2=string"
-VALUE_REGEX_COMMA_DELIM = _value_regex(",")
+VALUE_REGEX_COMMA_DELIM = _match_regex(match=".", delim=",")
+
+
+def _unquote(value):
+    r"""
+    Removes wrapping double quotes and any '\ ' characters. They are usually added to preserve spaces when passing
```

Comment:
You say "They are usually added". So does that mean there is a case that we wouldn't want to replace the double or '\ ' characters from the input?

Reply:
I dont think so, we will want to replace it. "Usually added" maybe a misnomer, preserving spaces inside quotes or even escaping quotes within quotes requires a `\` (escape) character.